jscomp/runtime/caml_obj.ml

```diff
@@ -133,12 +139,12 @@ let caml_lazy_make (fn : _ -> _) =
    In most cases, rec value comes from record/modules, 
    whose tag is 0, we optimize that case
 *)
-let caml_update_dummy : _ -> _ -> unit= [%raw{|function(x,y){
-  for (var k in y){
+let caml_update_dummy : _ -> _ -> unit= [%raw{|function (x, y) {
+  var set = function (k) {
     x[k] = y[k]
   }
-  }
-|}]
+  for_in(y, set)
+}|}]
```

Comment:
in general, it is not recommended to rely on closure for `raw` code (it is fragile), would you inline `for_in` here manually